api/src/pcapi/core/users/models.py

```diff
@@ -273,6 +273,13 @@ def checkPassword(self, passwordToCheck: str) -> bool:
     def get_notification_subscriptions(self) -> NotificationSubscriptions:
         return NotificationSubscriptions(**self.notificationSubscriptions or {})
 
+    def set_marketing_email_subscription(self, subscribe: bool) -> None:
+        self.notificationSubscriptions = (
+            {**self.notificationSubscriptions, "marketing_email": subscribe}
+            if self.notificationSubscriptions is not None
+            else {"marketing_email": subscribe}
+        )
```

Comment:
Petite proposition, utiliser le [union](https://peps.python.org/pep-0584/) pour simplifier :
```suggestion
        self.notificationSubscriptions = (self.notificationSubscriptions or {}) | {"marketing_email": subscribe}
```